src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -526,11 +527,14 @@ JvmtiVTMSTransitionDisabler::VTMS_vthread_start(jobject vthread) {
   assert(!thread->is_in_VTMS_transition(), "sanity check");
   assert(!thread->is_in_tmp_VTMS_transition(), "sanity check");
 
-  JvmtiEventController::thread_started(thread);
-  if (JvmtiExport::can_support_virtual_threads()) {
-    if (JvmtiExport::should_post_vthread_start()) {
-      JvmtiExport::post_vthread_start(vthread);
-    }
+  // If interp_only_mode is enabled then we must eagerly create JvmtiThreadState
+  // objects for globally enabled virtual thread filtered events. Otherwise,
+  // it is an important optimization to create JvmtiThreadState objects lazily.
+  if (JvmtiThreadState::seen_interp_only_mode()) {
+    JvmtiEventController::thread_started(thread);
+  }
+  if (JvmtiExport::should_post_vthread_start()) {
```

Comment:
Why is this check: `if (JvmtiExport::can_support_virtual_threads()) {` removed?

Reply:
It is because the `JvmtiExport::should_post_vthread_start()` returns `true` only if the `JvmtiExport::can_support_virtual_threads()` returns `true`.

The JVMTI `SetEventNotificationMode()` function checks for a required event capability by using the `JvmtiUtil::has_event_capability()` function that is defined in the jvmtiEnter.cpp generated with the `jvmtiEnter.xsl` script.

It looks like this:
```
// Check Event Capabilities
bool JvmtiUtil::has_event_capability(jvmtiEvent event_type, const jvmtiCapabilities* capabilities_ptr) {
  switch (event_type) {
    . . .
    case JVMTI_EVENT_VIRTUAL_THREAD_START:
      return capabilities_ptr->can_support_virtual_threads != 0;
    case JVMTI_EVENT_VIRTUAL_THREAD_END:
      return capabilities_ptr->can_support_virtual_threads != 0;
    . . .
```